Jint/ParseOptions.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System.Text.RegularExpressions;
+
+namespace Jint
+{
+    public interface IParseOptions
```

Comment:
I'm not a native speaker, but would `IParsingOptions` be more natura, also `ScriptParsingOptions` - just a thought

Reply:
I'm fine with either way, I guess we need an Oxford scholar for this 😆

Reply:
Naming things and cache invalidation, the two hard problems in software engineering :D 

I ran a quick search on the ASP.NET Core repo. Although there are some counter-examples but it's mostly `[...]ingOptions` or `[...]tionOptions`, so I'm convinced. :D